src/decorators/Directive.ts

```diff
@@ -0,0 +1,41 @@
+import { MethodAndPropDecorator } from "./types";
+import { SymbolKeysNotSupportedError } from "../errors";
+import { getMetadataStorage } from "../metadata/getMetadataStorage";
+import { DirectiveMetadata } from "../metadata/definitions/directive-metadata";
+
+export interface DirectiveArgs {
+  [arg: string]: any;
+}
+
+export function Directive(sdl: string): MethodAndPropDecorator & ClassDecorator;
+export function Directive(
+  name: DirectiveMetadata["nameOrSDL"],
+  args?: DirectiveMetadata["args"],
```

Comment:
Can we avoid the indirect types addressing and use type value (`Record<string, any>`) or named types (`DirectiveArgs`). It doesn't match the current code style and I see it everywhere in your PR.